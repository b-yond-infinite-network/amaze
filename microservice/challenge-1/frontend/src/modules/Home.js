import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";
import Container from "@material-ui/core/Container";

const useStyles = makeStyles((theme) => ({
  mainGrid: {
    marginTop: theme.spacing(3),
  },
  noticias: {
    flexGrow: 1,
    padding: 20,
  },
  nosotros: {
    flexGrow: 1,
  },
}));

export default function Landing() {
  const classes = useStyles();

  return (
    <React.Fragment>
      <CssBaseline />
      <Container maxWidth="lg">
      </Container>
    </React.Fragment>
  );
}
