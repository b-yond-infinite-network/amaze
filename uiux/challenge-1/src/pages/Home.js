import React, { PureComponent } from 'react'
import SearchBox from '../components/SearchBox';
import SongList from '../components/SongList';
import Lyrics from '../components/Lyrics';
import { withRouter } from 'react-router-dom'
import { connect } from 'react-redux'
import { requestSongList, requestLyric, requestSong } from '../reducers/songs'


class Home extends PureComponent {
  constructor (props){
    super(props);
    this.state = {}
  }

  handleArtistSelection = (e, {result}) => {
    {this.artistSelection(result.title)}
    {this.setState({ value: result.title })}
    {this.props.history.push({pathname: '/artist/' + result.title })}
  }
  artistSelection = artist => {
    this.SearchBoxChild.props.requestSongList(artist)
  }

  handleTrackSelection = trackID => {
    {this.lyricSelection(trackID)}
    {this.setState({ trackID: trackID })}
  }
  lyricSelection = trackID => {
    this.LyricsChild.props.handleOpenLyric(trackID)
  }

  handleOpenLyric = trackID => {
    this.LyricsChild.props.requestLyric(trackID);
    this.LyricsChild.props.requestSong(trackID);
    this.LyricsChild.setState({ open: true });
    
  }  
  render () {
    return (
      <div>
        <SearchBox 
          artistSelected = {this.state.value} 
          handleArtistSelection = {this.handleArtistSelection}/>
        <SongList 
          onRef={ref => (this.SearchBoxChild = ref)}
          handleTrackSelection={this.handleTrackSelection} 
          {...this.props}
        />
        <Lyrics 
          onRef={ref => (this.LyricsChild = ref)} 
          handleOpenLyric = {this.handleOpenLyric}
          />
      </div>
    )
  }
}

export default withRouter(
connect(
  ({ ui: { loading } }) => ({loading}), 
  {requestSongList, requestLyric, requestSong}
)
(Home))