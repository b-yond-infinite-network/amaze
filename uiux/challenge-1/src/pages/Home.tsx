import { RouteComponentProps } from "@reach/router";
import React, { useEffect, useState } from "react";
import List from "../features/List";
import { artistSearch } from "../shared/api/artist";
import Input from "../shared/components/Input";
import Loader from "../shared/components/Loader";
import useDebounce from "../shared/hooks/useDebounce";
import Artist from "../shared/types/artist";

const Home: React.FC<RouteComponentProps> = () => {
  const [value, setValue] = useState<string>("");
  const debouncedValue = useDebounce(value, 300);
  const [results, setResults] = useState<Artist[]>([]);
  const [loading, setLoading] = useState<boolean>(false);

  useEffect(() => {
    if (debouncedValue !== "") {
      setLoading(true);
      artistSearch(debouncedValue).then(response => {
        setResults(response);
        setLoading(false);
      });
    }
  }, [debouncedValue]);

  const onChange = (e: React.FormEvent<HTMLInputElement>) => {
    setValue(e.currentTarget.value);
  };

  return (
    <main data-testid="page-home">
      <h1>Karaoke needs words</h1>
      <Input value={value} onChange={onChange} placeholder="Type artist" />
      {!loading && results.length > 0 && (
        <List items={results} initialized={debouncedValue !== ""} />
      )}
      {loading && <Loader />}
    </main>
  );
};

export default Home;
