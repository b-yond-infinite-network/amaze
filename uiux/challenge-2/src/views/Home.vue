<template>
  <div class="home container">
    <img class="block-center" alt="Vue logo" src="@/assets/logo.png">
    <Welcome/>
  </div>
</template>

<script>
import Welcome from '@/components/Welcome.vue'

export default {
  name: 'home',
  components: {
    Welcome
  }
}
</script>

<style scoped lang="scss">
.home {
  min-height:90vh;
  margin:auto;
  display:flex;
  flex-direction:column;
  justify-content:center;
  align-items:center;
  justify-content:center;
}
</style>
