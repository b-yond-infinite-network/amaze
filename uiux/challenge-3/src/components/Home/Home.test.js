import Home from './index';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router'
import configureMockStore from 'redux-mock-store'
import middlewares from './../../store'

const initialState = {
    home: {
        tags: ['tag1', 'tag2']
    },
    common: {
        appName: 'test appName',
        token: '1234'
    }
};

describe('Home Component', () => {

    let mockStore;

    let tree;
    let wrapper;
    let render;

    let mockOnClickTag;

    beforeAll(() => {

        mockStore = configureMockStore(middlewares)(initialState);
        mockOnClickTag = jest.fn();

        const props = {
            tags: ['tag1', 'tag2'],
            onClickTag: mockOnClickTag
        };

        tree = <Provider store={mockStore}>
            <MemoryRouter initialEntries={["/"]}>
                <Home {...props} />
            </MemoryRouter>
        </Provider>;
        wrapper = mount(tree);
        console.log(wrapper.debug());
        render = renderer.create(tree).toJSON();

    });

    it('Should match the snapshot', () => {
        expect(render).toMatchSnapshot();
    });

    it('Should render without errors', () => {
        const component = wrapper.find('#home-page');
        expect(component.length).toBe(1);
    });

});