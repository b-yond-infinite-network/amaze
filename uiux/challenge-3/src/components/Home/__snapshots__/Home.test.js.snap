// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Should match the snapshot 1`] = `
<div
  className="home-page"
  id="home-page"
>
  <div
    className="container page"
  >
    <div
      className="row"
    >
      <div
        className="col-md-9"
        id="main-view"
      >
        <div
          className="feed-toggle"
        >
          <ul
            className="nav nav-pills outline-active"
          >
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href=""
                onClick={[Function]}
              >
                Your Feed
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href=""
                onClick={[Function]}
              >
                Global Feed
              </a>
            </li>
          </ul>
        </div>
        <div
          className="article-preview"
        >
          Loading...
        </div>
      </div>
      <div
        className="col-md-3"
      >
        <div
          className="sidebar"
        >
          <p>
            Popular Tags
          </p>
          <div
            className="tag-list"
            id="tag-list"
          >
            <a
              className="tag-default tag-pill"
              href=""
              onClick={[Function]}
            >
              tag1
            </a>
            <a
              className="tag-default tag-pill"
              href=""
              onClick={[Function]}
            >
              tag2
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
